File: Array_Functions/Reverse_an_Array_in_place/reverse_in_place.cpp
#include <iostream>
#include <vector>

using namespace std;

template <typename T>

void reverse_in_place(vector<T> &arr){
    int n = arr.size();
    for (int i = 0; i < n/2; i++){
        T temp = arr[i];
        arr[i] = arr[n-1-i];
        arr[n-1-i] = temp;
    }
    return;
}

int main()
{
    vector<int> arr = {1, 2, 3};
    reverse_in_place(arr);
    for (int num : arr)
        cout << num << " ";
    cout << endl;
    return 0;
}

File: Array_Functions/Reverse_an_Array_in_place/reverse_in_place.py
def reverse_in_place(arr):
    n = len(arr)
    for i in range(n // 2):
        temp = arr[i]
        arr[i] = arr[n-1-i]
        arr[n-1-i] = temp
    return

if __name__ == "__main__":
    arr = [1, 2, 3]
    reverse_in_place(arr)
    for num in arr:
        print(num, end=" ")
    print()

# You can also use reversed() built-in function
